### ISSUE TITLE
ToggleGroupControl: No focus style and selected style in Windows high contrast mode

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#50782]
### Description

Discovered while reviewing #50772

The ToggleGroupControl component has the following two problems in Windows high contrast mode:

- Cannot see that the control group is in focus
- It is impossible to know which one is selected

### Step-by-step reproduction instructions

- Enable Windows High Contrast mode.
- Access [the Storybook](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-togglegroupcontrol--default).
- Perform mouse and keyboard operations on the component.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/54422211/1462135e-ceb1-44e1-b832-a91d0cd4c484

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
